src/nacl/signing.py

```diff
@@ -97,17 +107,26 @@ def verify(self, smessage, signature=None, encoder=encoding.RawEncoder):
             signature and message concated together.
         :param signature: [:class:`bytes`] If an unsigned message is given for
             smessage then the detached signature must be provided.
-        :param encoder: A class that is able to decode the secret message and
-            signature.
+        :param encoder: (Deprecated) A class that is able to decode the secret
+            message and signature.
         :rtype: :class:`bytes`
         """
+        if encoder is None:
+            encoder = encoding.RawEncoder
+        else:
+            warn(("Automatic encoding/decoding of signed messages "
```

Comment:
This should be the same message (or better yet, this should be a function that everything invokes)